zarr/v3/group.py

```diff
@@ -107,73 +116,276 @@ def from_json(
         )
         return group
 
-    @classmethod
-    async def open_or_array(
-        cls,
-        store: StoreLike,
-        runtime_configuration: RuntimeConfiguration = RuntimeConfiguration(),
+    async def getitem(
+        self,
+        key: str,
     ) -> Union[Array, Group]:
-        store_path = make_store_path(store)
-        zarr_json_bytes = await (store_path / ZARR_JSON).get_async()
-        if zarr_json_bytes is None:
-            raise KeyError
-        zarr_json = json.loads(zarr_json_bytes)
-        if zarr_json["node_type"] == "group":
-            return cls.from_json(store_path, zarr_json, runtime_configuration)
-        if zarr_json["node_type"] == "array":
-            return Array.from_json(
-                store_path, zarr_json, runtime_configuration=runtime_configuration
+
+        store_path = self.store_path / key
+
+        if self.zarr_format == 3:
+            zarr_json_bytes = await (store_path / ZARR_JSON).get_async()
+            if zarr_json_bytes is None:
+                # implicit group?
```

Comment:
This is a bit dangerous because we have no good way of telling whether this path even exists.

Reply:
I agree and we should return to this. For now I'm going to put a log statement there and we can address this later.